baseplate/clients/kafka.py

```diff
@@ -0,0 +1,208 @@
+import logging
+import time
+
+from concurrent.futures import wait
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import confluent_kafka
+
+from confluent_kafka.avro import AvroProducer
+from confluent_kafka.avro import Producer
+from gevent.threadpool import ThreadPoolExecutor
+
+from baseplate import Span
+from baseplate.clients import ContextFactory
+from baseplate.lib import config
+
+logger = logging.getLogger(__name__)
+
+TOPIC_SUFFIX = "_avro"
+CHECK_INTERVAL_SECONDS = 0.001
+
+
+class AvroProducerContextFactory(ContextFactory):
+    def __init__(
+        self,
+        bootstrap_servers: str,
+        schema_registry: str,
+        acks: int,
+        thread_pool_max_size: int = 10,
+        prefetch_schema_ids: Optional[List[int]] = None,
+    ):
+        self.avro_producer = AvroProducer(
+            {
+                "bootstrap.servers": bootstrap_servers,
+                "schema.registry.url": schema_registry,
+                "acks": acks,
+                # force sending 1 message at a time--don't wait for larger batches
+                "queue.buffering.max.ms": 0,
+                "max.in.flight": 1,
+                # hardcode the defaults here because we reference them with our manual refreshing
+                "topic.metadata.refresh.interval.ms": 300000,
+                "metadata.max.age.ms": 900000,
+            }
+        )
+        self.threadpool = ThreadPoolExecutor(thread_pool_max_size)
+        self.initialize(self.threadpool, prefetch_schema_ids or [])
+        self.last_initialized_at = time.time()
+
+    def initialize(
+        self, threadpool: ThreadPoolExecutor, prefetch_schema_ids: List[int], block: bool = True
```

Comment:
I don't think we need to pass in the threadpool since it's already set on `self`, you can just access it there.